go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -108,15 +111,28 @@ func DemystifyPending(status v1.PodStatus) (pluginsCore.PhaseInfo, error) {
 			}
 
 		case v1.PodReasonUnschedulable:
-			// We Ignore case in which we are unable to find resources on the cluster. This is because
-			// - The resources may be not available at the moment, but may become available eventually
-			//   The pod scheduler will keep on looking at this pod and trying to satisfy it.
+			//  We Ignore case in which we are unable to find resources on the cluster unless system level
+			//  timeout is set. This is because The resources may be not available at the moment, but may become
+			//  available eventually The pod scheduler will keep on looking at this pod and trying to satistfy it
 			//
 			//  Pod status looks like this:
 			// 	message: '0/1 nodes are available: 1 Insufficient memory.'
 			//  reason: Unschedulable
 			// 	status: "False"
 			// 	type: PodScheduled
+
+			// If the pod is interruptible and is waiting to be scheduled for an extended amount of time,  it is possible there are
+			// no spot instances availabled in the AZ. In this case, we timeout with a system level error and will retry on a
+			// non spot instance AZ.
+			if val, ok := pod.ObjectMeta.Labels[Interruptible]; ok {
```

Comment:
Should we add a check here for `if c.Status == v1.ConditionTrue` or is Unschedulable a terminal condition that once entered, the pod will never exit?

Reply:
I found this comment for Unschedulable 
```	
// PodReasonUnschedulable reason in PodScheduled PodCondition means that the scheduler
// can't schedule the pod right now, for example due to insufficient resources in the cluster
```

However for the status that is being checked is the `ConditionStatus` and if it is in `ConditionTrue` then appears to mean that resource **is in** that condition.

```
// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
// can't decide if a resource is in the condition or not. In the future, we could add other
// intermediate conditions, e.g. ConditionDegraded.
```

Reply:
Added the check.